.github/workflows/build-containers.yml

```diff
@@ -21,174 +21,246 @@ jobs:
-               # root dir exists. Example value:
-               # "opensciencegrid/vo-frontend opensciencegrid/ospool-cm"
-               images=$(git diff --name-only \
-                                 "$BASE" \
-                                 "$GITHUB_SHA" |
-                        egrep "^$ORG_DIR/" |
-                        cut -d/ -f -2 |
-                        sort |
-                        uniq |
-                        xargs -I {} find . -type d \
-                                           -wholename ./{} \
-                                           -printf "%P\n")
-          else
-               # List all image root dirs. Example value:
-               # "opensciencegrid/vo-frontend opensciencegrid/ospool-cm"
-               images=$(find $ORG_DIR -mindepth 1 \
-                                      -maxdepth 1 \
-                                      -type d \
-                                      -printf "$ORG_DIR/%P\n")
-          fi
+            ORG_DIR=opensciencegrid
+            # Get the list of files changed based on the type of event
+            # kicking off the GHA:
+            # 1. For the main branch, diff the previous state of main vs
+            #    the current commit
+            # 2. For other branches (i.e., on someone's fork), diff main
+            #    vs the current commit
+            # 3. For PRs, diff the base ref vs the current commit
+            # 4. For everything else (e.g., dispatches), build all images
+            if [[ $GITHUB_EVENT_NAME == 'pull_request' ]] ||
+              [[ $GITHUB_EVENT_NAME == 'push' ]]; then
+                if [[ $GITHUB_EVENT_NAME == 'pull_request' ]]; then
+                    BASE=$(git merge-base origin/$GITHUB_BASE_REF HEAD)
+                elif [[ $GITHUB_REF == 'refs/heads/main' ]]; then
+                    BASE=${{github.event.before}}
+                else
+                    BASE=origin/main
+                fi
+                images=$(git diff --name-only \
+                                  "$BASE" \
+                                  "$GITHUB_SHA" |
+                          egrep "^$ORG_DIR/" |
+                          cut -d/ -f -2 |
+                          sort |
+                          uniq |
+                          xargs -I {} find . -type d \
+                                            -wholename ./{} \
+                                            -printf "%P\n")
+            else
+                images=$(find $ORG_DIR -mindepth 1 \
+                                        -maxdepth 1 \
+                                        -type d \
+                                        -printf "$ORG_DIR/%P\n")
+            fi
+            image_json=$(echo -n "${images:-dummy}" | jq -Rcs '.|split("\n") | map(select(. != ""))')
+            echo "$image_json" > image_list.json
+            echo "images=$(echo $images | tr '\n' ' ')" >> $GITHUB_OUTPUT
+      - name: Display image list
+        run: cat image_list.json
```

Comment:
You could just `echo "$image_json"` in the job step above instead of writing to disk then reading it off disk